src/services/mentors.js

```diff
@@ -1499,58 +1457,38 @@ module.exports = class MentorsHelper {
 			}
 
 			const mentorIds = extensionDetails.data.map((item) => item.user_id)
-			const userDetails = await userRequests.getUserDetailedList(mentorIds, tenantCode)
 
 			//Extract unique organization_codes
+			const uniqueOrganizations = [
+				...new Map(
+					extensionDetails.data.map((item) => [
+						item.organization_code,
+						{ organization_id: item.organization_id, organization_code: item.organization_code },
+					])
+				).values(),
+			]
+
 			const organizationCodes = [...new Set(extensionDetails.data.map((user) => user.organization_code))]
 
 			//Query organization table (only if there are codes to query)
-			let organizationDetails = []
-			if (organizationCodes.length > 0) {
-				const orgFilter = {
-					organization_code: {
-						[Op.in]: organizationCodes,
-					},
-				}
-				organizationDetails = await organisationExtensionQueries.findAll(orgFilter, tenantCode, {
-					attributes: ['name', 'organization_code', 'organization_id'],
-					raw: true, // Ensure plain objects
-				})
-
-				// Cache the organization details
-				if (organizationDetails && organizationDetails.length > 0) {
-					const cachePromises = organizationDetails
-						.map((org) => {
-							if (org.organization_code && org.organization_id) {
-								return cacheHelper.organizations
-									.set(tenantCode, org.organization_code, org.organization_id, org)
-									.catch((cacheError) => {
-										console.error(
-											`❌ Failed to cache organization ${org.organization_id} in mentor list:`,
-											cacheError
-										)
-									})
-							}
-						})
-						.filter(Boolean)
+			const orgMap = {}
+			if (uniqueOrganizations.length > 0) {
+				for (const orgData of uniqueOrganizations) {
+					let orgInfo = await cacheHelper.organizations.get(
+						tenantCode,
+						orgData.organization_code,
+						orgData.organization_id
+					)
 
```

Comment:
Organization cache fetch is sequential, can we make it parallelized with Promise.all .